scripts/test-with-kind.sh

```diff
@@ -36,105 +43,135 @@ check_is_installed kubectl "You can install kubectl with the helper scripts/inst
-  local __vpc_id=$( vpc_id )
-
-  for __type in ${INT_TEST_DIR}/*
-  do
-    case `basename $__type` in
-      test_app) # /test_app contains test app images
-        continue
-        ;;
-      sidecar)
-       APPMESH_PREVIEW=y AWS_ACCOUNT=$AWS_ACCOUNT_ID AWS_REGION=$AWS_REGION make helm-deploy WAIT_PROXY_READY=true
-       check_deployment_rollout appmesh-controller appmesh-system
-       kubectl get pod -n appmesh-system
-        ;;
-      sidecar-v1.22.2.0)
-       APPMESH_PREVIEW=y AWS_ACCOUNT=$AWS_ACCOUNT_ID AWS_REGION=$AWS_REGION make helm-deploy WAIT_PROXY_READY=true SIDECAR_IMAGE_TAG=v1.22.2.0-prod
-       check_deployment_rollout appmesh-controller appmesh-system
-       kubectl get pod -n appmesh-system
-        ;;
-    esac
-
-    echo -n "running integration test type $__type ... "
-    ginkgo -v -r $__type -- --cluster-kubeconfig=${KUBECONFIG} \
-            --cluster-name=${CLUSTER_NAME} --aws-region=${AWS_REGION} \
-            --aws-vpc-id=$__vpc_id
-    echo "ok."
-  done
-}
+  echo running tests
 
-function clean_up {
-    if [ -v "$TMP_DIR" ]; then
-        "${SCRIPTS_DIR}"/delete-kind-cluster.sh -c "$TMP_DIR" || :
-    fi
-    return
-}
+  kubectl delete ns appmesh-system || :
+  kubectl create ns appmesh-system
 
-trap "clean_up" EXIT
+  for __test in ${TEST_SUITES[@]}; do
+    local __test_dir="test/integration/$__test"
+    local __envoy_version="$ENVOY_LATEST_TAG"
 
-aws_check_credentials
+    # /test_app contains test app images, and is not itself an integration test
+    if [[ "$__test" == "test_app" ]]; then
+      continue
+    fi
 
-if [ -z "$AWS_ACCOUNT_ID" ]; then
-    AWS_ACCOUNT_ID=$( aws_account_id )
-fi
+    # This test specifically tests behavior for an older version of envoy
+    if [[ "$__test" == "sidecar-v1.22" ]]; then
+        __envoy_version="$ENVOY_1_22_TAG"
+    fi
 
-ecr_login $AWS_REGION $ECR_URL
+    # Try to delete the controller. Sometimes, carrying over controllers
+    # between tests can cause some minor issues.
+    helm delete -n=appmesh-system appmesh-controller || :
+
+    helm upgrade -i appmesh-controller config/helm/appmesh-controller --namespace appmesh-system \
+      --set image.repository="$CONTROLLER_IMAGE" \
+      --set image.tag="$CONTROLLER_TAG" \
+      --set sidecar.image.repository="$ENVOY_IMAGE" \
+      --set sidecar.image.tag="$__envoy_version" \
+      --set enableBackendGroups=true \
+      --set sidecar.waitUntilProxyReady=true \
+      --set region="$AWS_REGION" \
+      --set accountId="$AWS_ACCOUNT_ID" \
+      --set "env.AWS_DEFAULT_REGION='$AWS_REGION'" \
+      --set "env.AWS_ACCESS_KEY_ID='$AWS_ACCESS_KEY_ID'" \
+      --set "env.AWS_SECRET_ACCESS_KEY='$AWS_SECRET_ACCESS_KEY'" \
+      --set "env.AWS_SESSION_TOKEN='$AWS_SESSION_TOKEN'" \
+      --set sidecar.envoyAwsAccessKeyId="$AWS_ACCESS_KEY_ID" \
+      --set sidecar.envoyAwsSecretAccessKey="$AWS_SECRET_ACCESS_KEY" \
+      --set sidecar.envoyAwsSessionToken="$AWS_SESSION_TOKEN"
```

Comment:
Quick note on the permissions here:

This forwards the same set of credentials to two different locations: one, to the environment on the controller itself, and two, to the environment on the envoy containers. This could be finer grained - the envoy's need narrower permissions than the controller. This is needed as the runners lack the same AWS setup that let credentials be bootstrapped off the environment / metadata service.

I'm not thrilled by the system I added, and am very open to criticisms / suggestions here.

Reply:
I think this looks reasonable to me for now. Since you have the credentials as separate here, we can always introduce another role later.